### PR TITLE
Implement --page-load-timeout (closes #1645)

### LINKED ISSUE #1645: Add `--page-load-timeout`
(follow up #1521)
TestCafe starts tests when the `DomContentLoaded` event is raised. `window.load` event is raised with some delay sometimes.
Some pages executes page preparation scripts on the `window.load` event so we should wait for it. But it's possible you wouldn't like to wait additional timeouts so we should provide an ability to set this timeout to `0`.

It'll be 5 seconds by default.

Also add a per-test option via `setPageLoadTimeout(<ms>)`

### PR DESCRIPTION
**Pull request by AlexanderMoskovkin**
/cc @churkin @helen-dikareva @VasilyStrelyaev

### CODE REVIEW

File: src/client/core/utils/event.js [hunk 1]
```diff
@@ -67,7 +75,10 @@ export function documentReady () {
             nativeMethods.setTimeout.call(window, onContentLoaded, 1);
         else {
             bind(document, 'DOMContentLoaded', onContentLoaded);
-            bind(window, 'load', ready);
+            bind(window, 'load', () => {
```
Comment (inikulin) [inline src/client/core/utils/event.js]: I don't understand it. We will have ready called from `DOMContentLoaded` already (which always triggered before `window.onload`. I thought the algorithm will be:
1. Wait for `DOMContentLoaded`
2. Wait for `window.load` or `pageLoadTimeout` whichever comes first.

File: src/cli/argument-parser.js [hunk 2]
```diff
@@ -102,6 +102,7 @@ export default class CLIArgumentParser {
             .option('--app-init-delay <ms>', 'specify how much time it takes for the tested app to initialize')
             .option('--selector-timeout <ms>', 'set the amount of time within which selectors make attempts to obtain a node to be returned')
             .option('--assertion-timeout <ms>', 'set the amount of time within which assertion should pass')
+            .option('--page-load-timeout <ms>', 'set the amount of time within which TestCafe waits for the `window.load` event to fire before starting the test')
```
Comment (churkin) [inline src/cli/argument-parser.js]: Does it work only one time before starting the test?

Comment (VasilyStrelyaev) [inline src/cli/argument-parser.js]: Good question, I guess it works for all page loads in the test. Am I right, @AlexanderMoskovkin ?

Then it should be

```
            .option('--page-load-timeout <ms>', 'set the amount of time within which TestCafe waits for the `window.load` event to fire on page load before proceeding to the next test action')
```

Comment (AlexanderMoskovkin) [inline src/cli/argument-parser.js]: yes, you are right

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
/cc @churkin @helen-dikareva 
Please review but don't merge for now

File: src/test-run/index.js [hunk 3]
```diff
@@ -402,6 +411,11 @@ export default class TestRun extends Session {
         if (command.type === COMMAND_TYPE.assertion)
             return this._executeAssertion(command, callsite);
 
+        if (command.type === COMMAND_TYPE.setPageLoadTimeout) {
+            this.pageLoadTimeout = command.duration;
```
Comment (helen-dikareva) [inline src/test-run/index.js]: we've already done it in `_adjustConfigurationWithCommand` method and even return null [here](https://github.com/DevExpress/testcafe/pull/1648/files#diff-a21f56bee759db22062e32250e3df48cR403)

Comment (AlexanderMoskovkin) [reply src/test-run/index.js]: nice catch!

**Comment by AlexanderMoskovkin** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Comment by AlexanderMoskovkin** [comment]
@testcafe-build-bot \retest

**Pull request merged**